src/math/lp/random_updater_def.h

```diff
@@ -45,7 +44,9 @@ bool random_updater::shift_var(unsigned j) {
     if (ret) {
         const auto & A = m_lar_solver.A_r();
         for (const auto& c : A.m_columns[j]) {
-            m_var_set.erase(m_lar_solver.r_basis()[c.var()]);
+            unsigned k = m_lar_solver.r_basis()[c.var()];
+            if (m_var_set.contains(k))
```

Comment:
remove check to m_var_set.contains?

Reply:
Here is the remove method ` void remove(unsigned x) {
        SASSERT(contains(x));`